File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hashset = set(nums)

        biggestSoFar = 0

        for n in hashset:
            if n - 1 not in hashset:
                counter = 0
                while(n + counter in hashset):
                    counter += 1
                if counter > biggestSoFar:
                    biggestSoFar = counter
        return biggestSoFar


        

File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) 
    {
        unordered_set<int> hashset;
        for(int num : nums)
        {
            if(hashset.contains(num))
            {
                return true;
            }

            hashset.insert(num);
        }    
        return false;
    }
};